option.go

```diff
@@ -31,3 +31,9 @@ func WithTimeSkewLimit(skew time.Duration) Option {
 func WithMetadataSizeLimit(size int) Option {
 	return func(g *GoFakeS3) { g.metadataSizeLimit = size }
 }
+
+// WithIntegrityCheck enables or disables Content-MD5 validation when
+// putting an Object.
+func WithIntegrityCheck(check bool) Option {
```

Comment:
Ah oops!  I forgot to add the flag to the CLI, I've just added it now. I added this option just in case I got the implementation wrong - if it causes grief, doesn't work properly, etc, this gives people an out clause to disable it. If we keep improving the test cases, these things will make me a bit less nervous in the future but I'm just a wee bit wary at the moment.